vendor/github.com/livepeer/go-livepeer-basicnet/basic_network.go

```diff
@@ -78,6 +81,53 @@ func NewBasicVideoNetwork(n *NetworkNode) (*BasicVideoNetwork, error) {
 		mplChans:               make(map[string]chan *m3u8.MasterPlaylist),
 		transResponseCallbacks: make(map[string]func(transcodeResult map[string]string))}
 	n.Network = nw
+
+	//Set up a worker to write connections
+	if workDir != "" {
+		//Load connection information, try and connect to them.
+		bytes, err := ioutil.ReadFile(fmt.Sprintf("%v/conn", workDir))
+		if err == nil {
+			for _, line := range strings.Split(string(bytes), "\n") {
+				larr := strings.Split(line, "|")
+				if len(larr) == 2 {
+					addrs := strings.Split(larr[1], ",")
+					if err := nw.Connect(larr[0], addrs); err != nil {
```

Comment:
Something still feels weird to me about doing all this file IO mingled with the `nw.Connect()` function call inside the constructor for a Basic video Network object. I think the artchitecture is up to you, so you can go forward with what you think is best, but I could imagine clean pseudocode looking something like...

```
peerAddrs := peerCache.ReadPeersFromStore(workdir)
for peerAddr := range peerAddres {
   nw.Connect(peerAddr)
}

go peerCache.WriteUpdates(n)
```

And this way it would hide all the file IO stuff inside an object that's designed to interact with the persisted peer data, rather than dropping it into the constructor here.

Reply:
Yeah that makes sense.  We'll have to create `PeerCache` for https://github.com/livepeer/go-livepeer/issues/193